### ISSUE TITLE
v12 RC with Yarn: Unable to resolve "bson" from "node_modules/realm/dist/bundle.react-native.js"

### ISSUE THREAD

**Issue by Acetyld** [realm/realm-js#6040]
### How frequently does the bug occur?

Always

### Description

Unable to resolve "bson" from "node_modules/realm/dist/bundle.react-native.js"

### Can you reproduce the bug?

Always

### Reproduction Steps

Clean expo project, make a dev build, add     "realm": "^12.0.0-rc.3",

### Version

12.0.0-rc.3

### What services are you using?

-- select --

### Are you using encryption?

No

### Platform OS and version(s)

expo 49 0.72.3

### Build environment

Which debugger for React Native: ..

**Comment by takameyer** [comment]
Hi @Acetyld, I would really like to reproduce this so we can get this resolved.  Here are the steps I've have done:

```
npx create-expo-app bsonTest
cd bsonTest
npm install -D expo-dev-client
npm install realm
npx expo run:ios
```

This works without failure.  I have also added in the `App.tsx`

```
import Realm from "realm";

const realm = new Realm();
console.log(realm.schemaVersion); // outputs -1
```

This has worked for me with the node 16 and 18 as well as npm 8 and 9.  `bson` is marked as a peerDependency, and since version 7 of npm it should be automatically installed.  Can you verify what versions of npm and node you are using?
Also if I am missing any steps to reproduce this.  Thanks for helping us out!

**Comment by Acetyld** [comment]
Node: v18.17.0
NPM: 9.6.7

Also another thing i noticed, using newest expo dev client + realm was giving me the issue that the metro was stuck at 100% and i got a black screen, after a solid 2 hours of debuggin i noticed this only happend when flipper was open before starting metro load, if i disable flipper and then open app it works, after that i can safely open flipper
i reproduced this as well: https://github.com/Acetyld/realm-test-expo

**Comment by takameyer** [comment]
@Acetyld Alright, so this happens when upgrading, but not when installing realm into a clean project.  Thanks for the hint there.
We will have to investigate the flipper issue as well.  Can you create a new issue so we can track this?  Also providing the version of flipper you are using would be great.

**Comment by takameyer** [comment]
@Acetyld Actually it's not upgrading that is the issue.  It's `yarn`.  It appears `yarn` does not install `peerDependency` automatically.

**Comment by Acetyld** [comment]
@Acetyld correct! I am using yarn, 100% correct about that one.
Also: https://github.com/realm/realm-js/issues/6042 made it

**Comment by takameyer** [comment]
This will be fixed in the next release.  For now, `yarn add bson@4` would be the suggested workaround.

**Comment by Acetyld** [comment]
Perfect.
